src/support/lockedpool.cpp

```diff
@@ -200,8 +200,11 @@ void Win32LockedPageAllocator::FreeLocked(void* addr, size_t len)
 
 size_t Win32LockedPageAllocator::GetLimit()
 {
-    // TODO is there a limit on Windows, how to get it?
-    return std::numeric_limits<size_t>::max();
+    size_t minSet;
+    size_t maxSet;
+    DWORD flags;
+    GetProcessWorkingSetSizeEx(GetCurrentProcess(), &minSet, &maxSet, &flags);
```

Comment:
This needs error handling. I think logging a warning and returning 0 is good enough, but right now this will potentially return an invalid or uninitialized value in the case of an error.

Reply:
That Win32 function doesn't return an error code https://docs.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-getprocessworkingsetsizeex.